src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -20,51 +19,65 @@ internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProvider
         [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
         public async Task InlineTypeCheck1()
         {
-            await TestInRegularAndScriptAsync(
-@"class C
-{
-    void M()
-    {
-        [|var|] x = o as string;
-        if (x != null)
-        {
+            await TestCondition("x != null", "o is string x");
+            await TestCondition("null != x", "o is string x");
+
+            await TestCondition("(object)x != null", "o is string x");
+            await TestCondition("null != (object)x", "o is string x");
+
+            await TestCondition("x == null", "!(o is string x)");
```

Comment:
:memo: This test case replaces the original `TestMissingOnNullEquality`, showing new functionality.